src/app/dynamic-forms/cinchy-dynamic-forms.component.ts

```diff
@@ -140,14 +152,19 @@ export class CinchyDynamicFormsComponent implements OnInit, OnChanges, OnDestroy
       /* if (!isNullOrUndefined(this.RowId)) {
          this.spinner.show();
        }*/
-      this.getFormMetaData();
+      let expand=true;
+      await this.getFormMetaData(expand);
+      console.log(new Date(), 'start wait')
+      await new Promise(f => setTimeout(f, 2000));
```

Comment:
What's the purpose of this hard coded 2 second timeout here?

Reply:
we had a problem where both expanded and non expanded sections where loading parallelly. To avoid that, we had to wait the execution for non-expanded sections hence adding a timeout for 2 seconds,

Reply:
That's not the behaviour we want, we want to wait for the expanded sections to finish loading first, then start the non expanded sections

Reply:
we don't want to hardcode a 2 second wait time. You can probably do something with getFormMetaData to return a callback that tells us when it's done loading or create a service to keep track of what sections are loading/done and subscribe to those changes

Reply:
removed this hardcoded 2 second logic